### PR TITLE
Deal with version 0 messages.

### PR DESCRIPTION
**Pull request by smccauliff**
which just contain large message segments, but not headers of other kinds.

### CODE REVIEW

File: src/test/java/com/linkedin/kafka/clients/largemessage/Version0DeserializerTest.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+
+/**
+ * Tests that we can deserialize old encodings.
+ */
+public class Version0DeserializerTest {
+
+
+  @Test
+  public void testDeserializeVersion0Headers() {
```
Comment (becketqin) [inline src/test/java/com/linkedin/kafka/clients/largemessage/Version0DeserializerTest.java]: Should we test version 1 message format to ensure the combined deserializer works with the new format as well?

File: src/main/java/com/linkedin/kafka/clients/largemessage/Version0Deseriaizer.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+ * 1 byte   - version
+ * 4 bytes  - checksum to determine if bytes is large message segment or not.
+ * 16 bytes - messageId
+ * 4 bytes  - sequence number
+ * 4 bytes  - number of segments
+ * 4 bytes  - message size in bytes
+ * X bytes  - payload
+ * </pre>
+ */
+public class Version0Deseriaizer implements HeaderDeserializer {
```
Comment (smccauliff) [inline src/main/java/com/linkedin/kafka/clients/largemessage/Version0Deseriaizer.java]: ChainedDeserializer sounds too generic to me.

File: src/main/java/com/linkedin/kafka/clients/largemessage/Version0Deseriaizer.java [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+      throw new IllegalArgumentException("This deserializer only supports legacy serilized values.  No new values of"
+          + " this type are supported.");
+    }
+    _tryFirst = tryFirst;
+  }
+
+  @Override
+  public DeserializeResult deserializeHeader(ByteBuffer src) {
+    DeserializeResult deserializeResult = _tryFirst.deserializeHeader(src);
+    if (src == null && deserializeResult.value() == null) {
```
Comment (smccauliff) [inline src/main/java/com/linkedin/kafka/clients/largemessage/Version0Deseriaizer.java]: Yeah.  I've changed that.

File: src/main/java/com/linkedin/kafka/clients/largemessage/Version0Deseriaizer.java [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+      return deserializeResult;  //null value
+    }
+    if (deserializeResult.headers() != null || deserializeResult.value() != src) {
```
Comment (smccauliff) [inline src/main/java/com/linkedin/kafka/clients/largemessage/Version0Deseriaizer.java]: Yeah this is kind of hacky.  I've introduced a success flag in the returned result.

File: src/test/java/com/linkedin/kafka/clients/largemessage/Version0DeserializerTest.java (hunk 1 above)
Comment (smccauliff) [inline src/test/java/com/linkedin/kafka/clients/largemessage/Version0DeserializerTest.java]: Doing that too.

**Pull request closed without merging**